apicapi/apic_client.py

```diff
@@ -490,6 +519,28 @@ def _save_cookie(self, request, response):
             attributes = imdata[0]['error']['attributes']
         return attributes
 
+    def _calculate_signature(self, request, url, kwargs):
+        """ Sign the request using private key and put the result in cookies"""
+        if self._is_cert_auth():
```

Comment:
nit: should _do_request do this check before calling this instead? It's kind of odd that we always call this method for doing nothing when users use username and password